src/main.ts

```diff
@@ -2,77 +2,108 @@ import "./style.css";
 
-let fs: FileSystem | undefined;
+  let fs: FileSystem | undefined;
 
-switch (state?.scenario) {
-  case wn.Scenario.AuthCancelled:
-    console.log("Auth cancelled by user");
-    break;
+  switch (state?.scenario) {
+    case wn.Scenario.AuthCancelled:
+      console.log("Auth cancelled by user");
+      break;
 
-  case wn.Scenario.AuthSucceeded:
-  case wn.Scenario.Continuation:
-    // TODO: Maybe put some 'logged in as' state at the top right?
-    console.log("Logged in");
-    fs = state.fs;
-    break;
+    case wn.Scenario.AuthSucceeded:
+    case wn.Scenario.Continuation:
+      // TODO: Maybe put some 'logged in as' state at the top right?
+      console.log("Logged in");
+      fs = state.fs;
+      break;
 
-  case wn.Scenario.NotAuthorised:
-    wn.redirectToLobby(state.permissions);
-    console.log("Redirected to lobby");
-    break;
-}
+    case wn.Scenario.NotAuthorised:
+      wn.redirectToLobby(state.permissions);
+      console.log("Redirected to lobby");
+      break;
+  }
 
-const post = document.querySelector<HTMLInputElement>("#post")!;
+  const post = document.querySelector<HTMLInputElement>("#post")!;
 
-post.addEventListener("click", async function () {
-  console.log("attempting to save file");
-  const title = document.querySelector<HTMLInputElement>(".title-input")!;
-  // If title is blank, don't allow saving?
+  post.addEventListener("click", async function () {
+    console.log("attempting to save file");
+    const title = document.querySelector<HTMLInputElement>(".title-input")!;
+    // If title is blank, don't allow saving?
 
-  const body = document.querySelector<HTMLTextAreaElement>(".body-input")!;
-  // If body is blank, don't allow saving?
+    const body = document.querySelector<HTMLTextAreaElement>(".body-input")!;
+    // If body is blank, don't allow saving?
 
-  if (fs !== undefined) {
-    const filePath = wn.path.file("public", "Posts", `${title.value}.md`);
-    await fs.add(filePath, body.value).then(() => {
-      console.log("file saved");
-    });
-    await fs
-      .publish()
-      .then(() => {
-        console.log("file system published");
-      })
-      .catch((err) => {
-        console.log(err);
+    if (fs !== undefined) {
+      const filePath = wn.path.file("public", "Posts", `${title.value}.md`);
+      await fs.add(filePath, body.value).then(() => {
+        console.log("file saved");
       });
-  } else {
-    console.log("no file system");
-  }
+      await fs
+        .publish()
+        .then(() => {
+          console.log("file system published");
+        })
+        .catch((err) => {
+          console.log(err);
+        });
+    } else {
+      console.log("no file system");
+    }
+  });
+}
+
+const logIn = document.querySelector<HTMLInputElement>("#log-in")!;
+logIn.addEventListener("click", async function () {
+  // Initialize WebNative (which will force a log-in)
+  // TODO: Why does this not finish before the items below are hidden?
+  await main();
```

Comment:
Use the result of main function
<img width="870" alt="Screen Shot 2022-06-02 at 6 40 05 PM" src="https://user-images.githubusercontent.com/36746854/171770439-782eb4f4-d0a7-48d2-a09a-2e10b1f1e470.png">